features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/ComposeWireframeMapper.kt

```diff
@@ -121,50 +124,62 @@ internal class ComposeWireframeMapper(
     private fun createComposerWireframes(
         composer: Composer,
         wireframes: MutableList<MobileSegment.Wireframe>,
-        parentUiContext: UiContext
+        parentUiContext: UiContext,
+        internalLogger: InternalLogger
     ) {
         val compositionGroups = composer.compositionData.compositionGroups
         compositionGroups.forEach {
-            createCompositionGroupWireframes(it, wireframes, parentUiContext)
+            createCompositionGroupWireframes(it, wireframes, parentUiContext, internalLogger)
         }
     }
 
     private fun createCompositionGroupWireframes(
         compositionGroup: CompositionGroup,
         wireframes: MutableList<MobileSegment.Wireframe>,
-        parentUiContext: UiContext
+        parentUiContext: UiContext,
+        internalLogger: InternalLogger
     ) {
         var childrenUiContext = parentUiContext
         val composeContext = ComposeContext.from(compositionGroup)
         if (composeContext != null && !composeContext.name.isNullOrBlank()) {
-            val composeWireframe = mapCompositionGroup(compositionGroup, composeContext, childrenUiContext)
+            val composeWireframe =
+                mapCompositionGroup(compositionGroup, composeContext, childrenUiContext, internalLogger)
             if (composeWireframe != null) {
                 wireframes.add(composeWireframe.wireframe)
                 childrenUiContext = composeWireframe.uiContext ?: parentUiContext
             }
         }
 
         compositionGroup.compositionGroups.forEach {
-            createCompositionGroupWireframes(it, wireframes, childrenUiContext)
+            createCompositionGroupWireframes(it, wireframes, childrenUiContext, internalLogger)
         }
 
         compositionGroup.data.asSequence()
             .flatMap { getSubComposers(it) }
-            .forEach { createComposerWireframes(it, wireframes, childrenUiContext) }
+            .forEach { createComposerWireframes(it, wireframes, childrenUiContext, internalLogger) }
     }
 
     private fun mapCompositionGroup(
         compositionGroup: CompositionGroup,
         composeContext: ComposeContext,
-        parentUiContext: UiContext
+        parentUiContext: UiContext,
+        internalLogger: InternalLogger
     ): ComposeWireframe? {
         val mapper = composeMappers[composeContext.name] ?: return null
-        return mapper.map(compositionGroup, composeContext, parentUiContext)
+        return internalLogger.measureMethodCallPerf(
+            mapper.javaClass,
+            "$METHOD_CALL_MAP_PREFIX ${composeContext.name}",
+            METHOD_CALL_SAMPLING_RATE
+        ) {
+            mapper.map(compositionGroup, composeContext, parentUiContext)
+        }
     }
 
     // endregion
 
     companion object {
         private const val HUNDRED: Double = 100.0
+        private const val METHOD_CALL_MAP_PREFIX: String = "[Compose] Map with"
+        const val METHOD_CALL_SAMPLING_RATE = 1f
```

Comment:
Is there a reason to set this so high? It will send telemetry for every call to the mapper. You can use the existing [MethodCallSamplingRate](https://github.com/DataDog/dd-sdk-android/blob/develop/dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/metrics/MethodCallSamplingRate.kt) enum for the sampling rate

Reply:
Currently it's on feature branch of compose session replay, so it will be only me who can generate data when testing, and if I understand correctly `1f` means 1/100 of call instead of every call. 

And it seems `MethodCallSamplingRate` is not available yet in this feature branch. probably we will do a merge from master after Xavier comes back

Reply:
"1f means 1/100 of call instead of every call." - you're absolutely correct, my mistake.